src/attesters/sev-snp/collect_evidence.c

```diff
@@ -0,0 +1,105 @@
+/* Copyright (c) 2021 Intel Corporation
+ * Copyright (c) 2020-2021 Alibaba Cloud
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <linux/sev-guest.h>
+#include <rats-tls/log.h>
+#include <rats-tls/attester.h>
+#include "sev_snp.h"
+
+#define SEV_GUEST_DEVICE "/dev/sev-guest"
+
+static int snp_get_report(const uint8_t *data, size_t data_size, snp_attestation_report_t *report)
+{
+	struct snp_report_req req;
+	struct snp_report_resp resp;
+	struct snp_guest_request_ioctl guest_req;
+	snp_msg_report_rsp_t *report_resp = (struct msg_report_resp *)&resp.data;
+
+	if (data && (data_size > sizeof(req.user_data) || data_size == 0) || !report)
+		return -1;
+
+	/* Initialize data structures */
+	memset(&req, 0, sizeof(req));
+	req.vmpl = 1;
+	if (data)
+		memcpy(&req.user_data, data, data_size);
+
+	memset(&resp, 0, sizeof(resp));
+
+	memset(&guest_req, 0, sizeof(guest_req));
+	guest_req.msg_version = 1;
+	guest_req.req_data = (__u64)&req;
+	guest_req.resp_data = (__u64)&resp;
+
+	/* Open the sev-guest device */
+	int fd = open(SEV_GUEST_DEVICE, O_RDWR);
+	if (fd == -1) {
+		RTLS_ERR("failed to open %s\n", SEV_GUEST_DEVICE);
+		return -1;
+	}
+
+	/* Issue the guest request IOCTL */
+	int ret = ioctl(fd, SNP_GET_REPORT, &guest_req);
+	if (ret == -1) {
+		RTLS_ERR("failed to issue SNP_GET_REPORT ioctl, firmware error %llu\n",
+			 guest_req.fw_err);
+		goto out_close;
+	}
+
+	/* Check that the report was successfully generated */
+	if (report_resp->status != 0) {
+		RTLS_ERR("firmware error %#x\n", report_resp->status);
+		goto out_close;
+	} else if (report_resp->report_size > sizeof(*report)) {
```

Comment:
change to
```
if (report_resp->status != 0) {
    ...
    goto out_close;
}

if (report_resp->report_size > sizeof(*report)) {
    ...
}
```

Reply:
please change to if (report_resp->report_size != sizeof(*report))